src/webui/api/torrentscontroller.cpp

```diff
@@ -1414,3 +1414,19 @@ void TorrentsController::renameFolderAction()
         throw APIError(APIErrorType::Conflict, error.message());
     }
 }
+
+void TorrentsController::exportAction()
+{
+    requireParams({u"hash"_qs});
+
+    const auto id = BitTorrent::TorrentID::fromString(params()[u"hash"_qs]);
+    const BitTorrent::Torrent *const torrent = BitTorrent::Session::instance()->findTorrent(id);
+    if (!torrent)
+        throw APIError(APIErrorType::NotFound);
+
+    const nonstd::expected<QByteArray, QString> result = torrent->exportToBuffer();
+    if (!result)
+        throw APIError(APIErrorType::Conflict, tr("Unable to export torrent file"));
```

Comment:
(Not sure if anyone wants it) in case you want to reveal the real error to user:
```suggestion
        throw APIError(APIErrorType::Conflict, tr("Unable to export torrent file. Error: %1").arg(result.error()));
```
or even include the torrent name:
```suggestion
        throw APIError(APIErrorType::Conflict, tr("Unable to export torrent file. Torrent: %1. Error: %2").arg(torrent->name(), result.error()));
```

Reply:
Will include the original error 👍 . Going to omit the torrent name since this API only operates against a single torrent.